### PR TITLE
DeathChase3D [recipe]

### PR DESCRIPTION
**Pull request by bach5000**
The game's original website: http://osgameclones.com/
Do you guys want me to add this game to HaikuArchives? This game runs perfectly fine on Haiku. I also have the Git repo here: https://github.com/bach5000/DeathChase3D

### CODE REVIEW

**Comment by Begasus** [comment]
I think it would be better to place it in "games/action" : http://gpo.zugaina.org/games-action/deathchase3d

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++
+	cmd:make
+	"
+
+BUILD()
+{
+	cd deathchase3d
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```suggestion
	cd deathchase3d
```

File: haiku-games/deathchase3d/patches/deathchase3d-0.9.patchset [hunk 2]
```diff
@@ -0,0 +1,165 @@
+From 76ac1d69a58ca0d39edf67307f2f71e837e13ae2 Mon Sep 17 00:00:00 2001
+From: Bach Nguyen <bach5000@gmail.com>
+Date: Sun, 11 Nov 2018 23:03:20 +0000
+Subject: [PATCH] Makefile + minor changes
```
Comment (alaviss) [inline haiku-games/deathchase3d/patches/deathchase3d-0.9.patchset]: Why is this necessary when the source provides a `./configure` already?

**Comment by alaviss** [comment]
And I don't think we should add this to HaikuArchives, since it's not a BeOS/Haiku-exclusive title.

**Comment by bach5000** [comment]
@alaviss @Begasus I'll apply the changes in a moment :)

File: haiku-games/deathchase3d/patches/deathchase3d-0.9.patchset (hunk 2 above)
Comment (bach5000) [reply haiku-games/deathchase3d/patches/deathchase3d-0.9.patchset]: Will investigate... I honestly have no idea why I added a makefile :)
Maybe because I was afraid the generated makefile wouldn't have correct directories for Haiku.

**Comment by bach5000** [comment]
@alaviss Okay, tried building with the configure file, but then the build crashes :(
We might want to keep the Makefile for now, not only because the game builds with it, but also it has a nice Haiku Makefile format!

**Comment by alaviss** [comment]
> @alaviss Okay, tried building with the configure file, but then the build crashes :(
> We might want to keep the Makefile for now, not only because the game builds with it, but also it has a nice Haiku Makefile format!

Can I have the log? AFAIK @Begasus has looked into building with the configure script, but I'm not sure if he succeeded.

And I'm not fond of writing custom Makefiles. You might have missed some installation steps, and/or missed some configuration options that might be needed.

**Comment by bach5000** [comment]
I'm illiterate in Makefile/configure language, so if anyone could identify where this error is coming from, that'll be great!
@alaviss The terminal output:
```
Welcome to the Haiku shell.

~/Desktop/deathchase3d-0.9> ./configure 
checking build system type... admin/config.guess: unable to guess system type

This script, last modified 2002-01-23, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

    ftp://ftp.gnu.org/pub/gnu/config/

If the version you run (admin/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2002-01-23

uname -m = x86_64
uname -r = 1
uname -s = Haiku
uname -v = hrev52295+96 Sep 27 2018 18:48:

/usr/bin/uname -p = 
/bin/uname -X     = 

hostinfo               = 
/bin/universe          = 
/usr/bin/arch -k       = 
/bin/arch              = 
/usr/bin/oslevel       = 
/usr/convex/getsysinfo = 

UNAME_MACHINE = x86_64
UNAME_RELEASE = 1
UNAME_SYSTEM  = Haiku
UNAME_VERSION = hrev52295+96 Sep 27 2018 18:48:
configure: error: cannot guess build type; you must specify one
~/Desktop/deathchase3d-0.9> 
```

**Comment by pulkomandy** [comment]
Replace config.guess and config.sub with up to date versions, you cun do this using autoreconf -vfi

**Comment by Begasus** [comment]
Even after several attempts (with libtoolize, autoconf, autoreconf) I ended up with "version mismatch for libtool" (with configure and make)

**Comment by pulkomandy** [comment]
Since this code is unmaintained and it is unlikely that we have to merge things with upstream, I'd say going with the makefile engine is simpler than trying to understand deprecated autotools mess.

**Comment by alaviss** [comment]
I figured out the solution actually.

1. Modify `Makefile.am`: remove any occurrence of `libtool.m4.in`
2. Regenerate `acinclude.m4`: `cat admin/acinclude.m4.in special.m4.in > acinclude.m4`
3. Run `autoreconf -vfi`

**Comment by Begasus** [comment]
Not sure what the game play is about (haven't tested fully) but not my cup of tea ;) (but the suggestion from @alaviss works) :)

**Comment by bach5000** [comment]
@alaviss @Begasus @pulkomandy Updated the recipe to use autoconf instead 😄 
Builds on x86_64.

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libsdl
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```diff
-	makefile_engine
-	cmd:aclocal
-	cmd:autoconf
-	cmd:automake
+	cmd:autoreconf
```

You should only specify what's required.

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	cmd:ld
+	cmd:libtoolize
+	cmd:g++
+	cmd:make
+	"
+
+BUILD()
+{
+	autoreconf -vfi
+	runConfigure ./configure --enable-shared
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```suggestion
	runConfigure ./configure --bindir="$appsDir"
```

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	make
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```suggestion
	make $jobArgs
```

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+INSTALL()
+{
+   	mkdir -p $appsDir
+	cp deathchase3d/deathchase3d $appsDir
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```diff
-  	mkdir -p $appsDir
-	cp deathchase3d/deathchase3d $appsDir
+	make install
```

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	makefile_engine
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
+	cmd:ld
+	cmd:libtoolize
+	cmd:g++
+	cmd:make
+	"
+
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]: Instead of the insanely huge patchset, we could write a `PATCH()` function to modify the files automatically.

    sed -E 's|(\$\(top_srcdir\)/)?admin/libtool.m4.in||g' -i Makefile.am

This should produce the desired changes on `Makefile.am`

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+BUILD_PREREQUIRES="
+	cmd:ld
+	cmd:libtoolize
+	cmd:g++
+	cmd:make
+	"
+
+PATCH()
+{
+	sed -E 's|(\$\(top_srcdir\)/)?admin/libtool.m4.in||g' -i Makefile.am
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]: You can add the code that deals with regenerating `acinclude.m4` and then the entire patchset could be dropped.

**Comment by bach5000** [comment]
@alaviss Done!

File: haiku-games/deathchase3d/deathchase3d-0.9.recipe [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	haiku
+	lib:libsdl
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libsdl
+	"
+BUILD_PREREQUIRES="
+	cmd:ld
```
Comment (alaviss) [inline haiku-games/deathchase3d/deathchase3d-0.9.recipe]:
```suggestion
	cmd:autoreconf
	cmd:ld
```

Remember, always try to build your recipe after every change ;)

Comment (bach5000) [reply haiku-games/deathchase3d/deathchase3d-0.9.recipe]: @alaviss Hehe sorry! I couldn't test the changes on a school computer: I was just pushing the commits first so I could test it later :)
Will add `cmd:autoreconf` and the docs once I come home.

For the docs, do I install them to $docDir? Something like `cp deathchase3d/docs/en $docDir`?

**Comment by bach5000** [comment]
@alaviss Hehe sorry! I couldn't test the changes on a school computer: I was just pushing the commits first so I could test it later :)
Will add cmd:autoreconf and the docs once I come home.

For the docs, do I install them to `$docDir`? Something like `cp deathchase3d/docs/en $docDir`?

**Comment by alaviss** [comment]
> For the docs, do I install them to `$docDir`? Something like `cp deathchase3d/docs/en $docDir`?

Yes, something like that

While you're at it, please move the category to "games-action". "haiku-*" only applies for BeOS/Haiku-exclusive applications
